elastic/kibana
# [Cases] Notify assignees when assigned to a case

## Summary

The PR adds the ability to notify users by email when assigned to a case. A user is:
- Not notified if he/she assigns themselves
- Notified if added as an assignee to a case
- Not notified if removed from a case

I did not add integration tests due to the complexity of simulating an email server. I added unit test coverage. If integration test coverage is needed we can add the tests on another PR.

Depends on: https://github.com/elastic/kibana/pull/143303
Fixes: https://github.com/elastic/kibana/issues/142307

## Email screenshot

<img width="365" alt="Screenshot 2022-11-08 at 7 05 39 PM" src="https://user-images.githubusercontent.com/7871006/200629737-86e56092-bff7-4fee-954a-d925c8b0c3d3.png">


@shanisagiv1 @lcawl What do you think about the content of the email (see screenshot)?

## Testing

1. Put the following in your `kibana.yml`:

```
notifications.connectors.default.email: 'mail-dev'

xpack.actions.preconfigured:
  mail-dev:
    name: preconfigured-email-notification-maildev
    actionTypeId: .email
    config:
      service: other
      from: mlr-test-sink@elastic.co
      host: localhost
      port: 1025
      secure: false
      hasAuth: false
```

2. Install [`maildev`](https://www.npmjs.com/package/maildev): `npm install -g maildev`
3. Run `maildev`: `maildev`
4. Open MailDev's web interface (http://0.0.0.0:1080/)
5. Create a case and assign users. You should see the emails in the MailDev inbox. 
6. Update the assignees of a case. You should see the emails in the MailDev inbox. 

Note: If you assign yourself you should not see an email. If you delete an assignee you should not see an email.

### Checklist

Delete any items that are not applicable to this PR.



### For maintainers


## Release notes

Notify users by email when assigned to a case
